### ISSUE TITLE
How to compile it.

### ISSUE THREAD

**Issue by elotronico** [zavg/linux-0.01#2]
Hello, 
I woluld to compile it, can you indicate me what compiler an linker can i use??

Thanks

**Comment by ghost** [comment]
run tools/build.c

**Comment by JL2210** [comment]
@elotronico Run `make`.

**Comment by jlagarespo** [comment]
@elotronico as @JL2210 said, just install the dependencies and run ``make``. This is such an early version of the kernel that the dependencies are minimal. Just a basic C crosscompiler, the GNU assembler, and for the linking the GNU linker should do the work just fine. If you are using let's say, debian, or ubuntu itself, you should have most tools already preinstalled.

**Comment by JL2210** [comment]
Yeah. But you can only build it on a 32-bit version of Minix or Linux, so
keep that in mind.

**Comment by jlagarespo** [comment]
I mean, you can build in windows, but you'll need to install a ton of things that already come with most linux distros.

**Comment by JL2210** [comment]
No, you can't. Windows uses PE, which is incompatible with a.out, COFF, and
ELF. If you can build this on Windows, I will be VERY surprised.

**Comment by jlagarespo** [comment]
I mean, you'd have to use a weird toolchain that is compatible with Windows, and that'd make no sense at all, having linux. But if you manage to get such thing there should be nothing stopping you from having a PE compiler that builds to ELF, COFF and such things. NASM itself for windows has an option for ELF, and im sure that with a bit of research you should be able to find a windows C crosscompiler for ELF, COFF, and as you call it "a.out". Thought of course, Linux is much better for this things, as you have everything already being native which makes thing 500% easier to build.

**Comment by JL2210** [comment]
Yeah.

FYI: `a.out` is a SUPER OLD binary format that was used by Linux until kernel ~2.0 or so, at which point it was replaced by ELF. It doesn't support 64-bit binaries, and at this point the kernel makers are phasing it out (kernel 5.1 will supposedly have support removed totally).

**Comment by JL2210** [comment]
But the main reason I mentioned 32-bit Minix and Linux is because the sources require all the tools to be 32-bit. On a 64-bit system, you have to do a whole bunch of stuff (editing the `Makefile`s, editing source code, and removing old compiler flags) to get it to build, much less work.

**Comment by jlagarespo** [comment]
Wait isn't there also windows for x86_32?

**Comment by JL2210** [comment]
I'll let you figure that out :).

**Comment by jlagarespo** [comment]
[There seems to be](https://support.microsoft.com/en-us/help/15056/windows-32-64-bit-faq1), thought my computer (at home, where I am not rn) is 64 bit.
Thought of course, all this conversation doesn't really matter that much, having access to Linux and all it's tools.

**Comment by JL2210** [comment]
Don't worry, there is. Microsoft wouldn't lose out on that big of a market.

**Comment by jlagarespo** [comment]
Now with all this conversation I want to try to build it on windows, if I have time later i might try to do so. If i can't build in my host then i'll use a VM. (if I remember to try it, because considering my brain's incredibly powerful memory I am not so sure about it ~~sarcasm detected~~)

**Comment by elotronico** [comment]
Thanks all for your replies!!!!

This weekend I will try to compile it in an old 32 bits computer ;)

Linux rocks!!
